src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,101 @@
+package duke.storage;
+
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * The object to read and store the file.
+ */
+public class Storage {
+    private String filePath; // The file's path.
+
+    /**
+     * Initiates the object.
+     *
+     * @param filePath The file's path.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads the file.
+     *
+     * @return Tasks loaded from the file.
+     * @throws DukeException When cannot found file or format incorrect.
+     * @throws FileNotFoundException Would not happen.
+     */
+    public ArrayList<Task> load() throws DukeException, FileNotFoundException {
+        ArrayList<Task> tasks = new ArrayList<Task>();
+
+        File f = new File(filePath);
+        if (!f.exists()) {
+            throw new DukeException("File not found.");
```

Comment:
In **Storage** class, maybe you could make the class to automatically create the data file for the user if it hasn't existed? This can make it easier for the user to start using the app. Also, maybe you could abstract away the part where you check whether the task is done and mark them as done? Cuz there's repetition and can make it quite hard to adapt to changes since you have to change all the code at once. Maybe you could put this part after the switch statement.